README.md

```diff
@@ -181,16 +192,45 @@ final class Gender extends Enum
 }
 ```
 
-Get an instance of your enum type:
+The `AutoDiscoveredValuesTrait` also allows you to discover values from other classes.
+Given the following class holding constants:
 
 ```php
 <?php
-$enum = Gender::get(Gender::Male);
+
+namespace MangoPay;
+
+final class EventType
+{
+    const KycCreated = "KYC_CREATED";
+    const KycSucceeded = "KYC_SUCCEEDED";
+    const KycFailed = "KYC_FAILED";
+}
 ```
 
-You can easily retrieve the enumeration's value by using `$enum->getValue();`
+You can create an enum from it by overriding the `AutoDiscoveredValuesTrait::getDiscoveredClasses()` method:
 
-> 📝 Enum values are supposed to be integers or strings.
+```php
+<?php
+
+namespace App\Enum;
+
+use Elao\Enum\Enum;
+use Elao\Enum\AutoDiscoveredValuesTrait;
+
+final class MangoPayEventType extends Enum
+{
+    use AutoDiscoveredValuesTrait;
+
+    protected static function getDiscoveredClasses(): array
+    {
+        return [self::class, MangoPay\EventType::class];
+    }
+}
+
+# Usage:
+MangoPayEventType::get(MangoPay\EventType::KycCreated);
```

Comment:
does `MangoPayEventType::KycCreated()` still work?

Reply:
No, as the contract for this magic calls to work is to have a matching constant in the enum class.
It could be improved though.